EapWebInterface/src/main/java/de/hpi/unicorn/application/pages/input/generator/GeneratePanel.java

```diff
@@ -242,4 +307,124 @@ private void addSubmitButton() {
         final Button submitButton = new Button("submitButton");
         layoutForm.add(submitButton);
     }
+
+    /**
+     * Add a button to export the input values given via input fields
+     */
+    private void addExportValuesButton() {
+        final AjaxLink exportButton = new AjaxLink<Void>("exportValuesButton") {
+            @Override
+            public void onClick(AjaxRequestTarget target) {
+                layoutForm.modelChanged();
+                final AJAXDownload jsonDownload = new AJAXDownload() {
+                    @Override
+                    protected IResourceStream getResourceStream() {
+                        Map<TypeTreeNode, String> inputsForExporter = new HashMap<>();
+                        for (AttributeInput input : attributeInputs) {
+                            inputsForExporter.put(input.getAttribute(), input.getInput());
+                        }
+                        final File json = JsonExporter
+                                .generateExportFileWithValues(selectedEventType, inputsForExporter, eventCount, scaleFactor, eventTimestamps);
+                        if (json == null) {
+                            return null;
+                        }
+                        return new FileResourceStream(new org.apache.wicket.util.file.File(json));
+                    }
+                    @Override
+                    protected String getFileName() {
+                        return selectedEventType.getTypeName() + "-values.json";
+                    }
+                };
+                GeneratePanel.this.add(jsonDownload);
+                jsonDownload.initiate(target);
+
+                GeneratePanel.this.page.getFeedbackPanel().success("Json created.");
+                target.add(GeneratePanel.this.page.getFeedbackPanel());
+            }
+        };
+        layoutForm.add(exportButton);
+    }
+
+    /**
+     * Add a upload field to upload a file.
+      */
+    private void addImportField() {
+        uploadField = new FileUploadField("importValuesUpload");
+        this.importForm.add(uploadField);
+    }
+
+    /**
+     * Add a button to import values from a file.
+     */
+    private void addImportSubmitButton() {
+        AjaxButton button = new AjaxButton("importValuesButton") {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Reduce this anonymous class number of lines from 59 to at most 20, or make it a named class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://bpt-lab.org/sonarqube/coding_rules#rule_key=squid%3AS1188)